templates/refresh_facts.erb

```diff
@@ -11,10 +11,31 @@ opt.on("--out FILE", "-o", "Path to facts") do |v|
   @outfile = v
 end
 
+opt.on("--pid FILE", "-p", "Path to pidfile") do |v|
+  @pid = v
+end
+
 opt.parse!
 
 abort("Please specify a file to write to") unless @outfile
 
+if @pid
+  begin
+    lock = File.open(@pid, File::CREAT | File::EXCL)
+    lock.write($$)
+
+    at_exit do
+      begin
+        lock.close
+      ensure
+        File.delete(@pid)
+      end
+    end
+  rescue
+    abort("PID file already exists, is MCollective refresh facts already running?")
```

Comment:
My thought is that the only time this should happen, is if either something is horribly broken, or if the refresh interval is far too low.
Either way, a mail about it seems mostly reasonable.